pkg/controller/machine/machine.go

```diff
@@ -890,6 +913,15 @@ func (r *Reconciler) getNode(instance instance.Instance, provider providerconfig
 		}
 		for _, nodeAddress := range node.Status.Addresses {
 			for instanceAddress := range instance.Addresses() {
+				// Hetzner reuses IP addresses, so if we rollout a Machine, it can happen that a new
```

Comment:
Every cloud provider reuses public IP addresses when released :-) This is a mitigation for when the providerID is not specified in the node and we fallback to the logic based on IP matching. I think that the comment could be clearer.

Reply:
Fixed.